lib/container/ModuleFederationPlugin.js

```diff
@@ -22,15 +22,17 @@ class ModuleFederationPlugin {
 	 */
 	apply(compiler) {
 		const { options } = this;
+		const { output } = compiler.options;
 		new ContainerPlugin({
 			name: options.name,
-			library: options.library,
+			library: options.library || options.library.type || output.library.type,
```

Comment:
```suggestion
			library: options.library,
```

undefined will use the global defined library